test/jdk/java/awt/Mouse/MouseModifiersUnitTest/ExtraButtonDrag.java

```diff
@@ -22,110 +22,121 @@
+            Dimension size = frame.getSize();
+            centerFrame = new Point(location.x + size.width / 2,
+                location.y + size.height / 2);
+            outboundsFrame = new Point(location.x + size.width * 3 / 2,
+                location.y + size.height / 2);
+        });
+
+        System.out.println("areExtraMouseButtonsEnabled() == "
+            + Toolkit.getDefaultToolkit().areExtraMouseButtonsEnabled());
+
+        for (int i = 0; i < MouseInfo.getNumberOfButtons(); i++) {
+            System.out.println("button to drag = " + (i + 1)
+                + " : value passed to robot = " + buttonMask[i]);
+
+            try {
+                dragMouse(buttonMask[i], centerFrame.x, centerFrame.y,
+                    outboundsFrame.x, outboundsFrame.y);
+            } catch (IllegalArgumentException e) {
+                throw new RuntimeException("Test failed. Exception occured.",
+                    e);
+            }
 
-                robot.delay(500);
-                //this is a choice-case for X protocol issue: native events from extra buttons doesn't contain
-                // the correct state so it's unable to decide if there is a drag or move. By default we send MOVED event.
-                //XToolkit: extra buttons should report MOVED events only
-                //WToolkit: extra buttons should report DRAGGED events only
-                if (i > 2){ //extra buttons only
-                    if (tk.equals("sun.awt.X11.XToolkit")) {
-                        if (!moved || dragged) {
-                            throw new RuntimeException("Test failed."+ tk +" Button = " +(i+1) + " moved = "+moved +" : dragged = " +dragged);
-                        }
-                    } else { //WToolkit
-                        if (moved || !dragged) {
-                            throw new RuntimeException("Test failed."+ tk +" Button = " +(i+1) + " moved = "+moved +" : dragged = " +dragged);
-                        }
+            // this is a choice-case for X protocol issue: native events from
+            // extra buttons doesn't contain
+            // the correct state so it's unable to decide if there is a drag or
+            // move. By default we send MOVED event.
+            // XToolkit: extra buttons should report MOVED events only
+            // WToolkit: extra buttons should report DRAGGED events only
+            if (i > 2) { // extra buttons only
+                if (tk.equals("sun.awt.X11.XToolkit")) {
+                    if (!moved || dragged) {
+                        throw new RuntimeException("Test failed." + tk
+                            + " Button = " + (i + 1) + " moved = " + moved
+                            + " : dragged = " + dragged);
                     }
-                } else {
-                    if (moved || !dragged){
-                        throw new RuntimeException("Test failed. Button = " +(i+1) + " not dragged.");
+                } else { // WToolkit
```

Comment:
And macOS ?
Seems like this should be (1) X, (2) everything else ?
And I think that checking the Toolkit like this is iffy.
When we get to Wayland it will be WLToolkit on Linux and the test will do the wrong thing.
Yes, I am assuming that this aspect of Wayland will be X compatible ...
This is pre-existing so I leave it up to you but I'd prefer to have a check that doesn't use the name of an implementation class if there's an alternative. Like just checking if this is Linux.